rfcs/ALL/SHM.md

```diff
@@ -0,0 +1,58 @@
+# Zenoh SHM
+
+**Supported platforms:**
+- Linux
+- MacOS (and other BSD)
+- Windows
+
+**API support:**
+- Rust
+- C
+- C++
+
+## General Concepts
+
+Zenoh SHM is designed to provide true zero-copy optimization abstracted behind the classic ZBytes API. The basic SHM data buffer type is `ZShm` (or it's mutable representation `ZShmMut`) which is typically wrapped into ZBytes, providing smooth integration into generalized application code:
+```
+________________________________                       ____________________________________
+|                              |                       |                                  |
+|        SHM Publisher         |     _____________     |     Non-SHM-aware Subscriber     |
+|                              |     |           |     |                                  |
+| alloc() -> ZShmMut -> ZBytes-|---->| Localhost |-----|-> ZBytes (implicitly wraps ZShm) |
+|______________________________|     |   Zenoh   |     |__________________________________|
+                                     |  Network  |                   
+                                     |___________|     ______________________________________
+                                           |           |                                    |
+                                           |           |        SHM-aware Subscriber        |
+                                           |           |                                    |
+                                           |-----------|--> ZBytes (implicitly wraps ZShm)  |
+                                                       | convertable to ZShm and/or ZShmMut |
+                                                       |____________________________________|
+```
+
+There are no limits on SHM buffer lifetime, number of shallow copies, number of republications, number of hops or network topology. SHM buffer is not pinned to particular Zenoh Session - it is possible to publish the same buffer multiple times through different Sessions. SHM buffer can be used anywhere where ZBytes is used.
+
+Zenoh Sessions probe and negotiate on SHM support. For participants not supporting SHM (because of their config, compilation flags, access rights or non-localhost location) any published SHM buffer will be implicitly converted into non-SHM one at the last hop before leaving SHM Domain boundary.
+
+SHM buffers are reference counted with additional mechanics applied to support dangling references recovery (in case of process holding SHM buffer crashes) to keep system robust.
```

Comment:
```suggestion
SHM buffers are reference counted with additional mechanics applied to support dangling reference recovery (in case a process holding an SHM buffer crashes) to keep the system robust.
```